example/no_tls_both.cpp

```diff
@@ -172,35 +196,34 @@ void server_proc(Server& s, std::set<con_sp_t>& connections, mi_sub_con& subs) {
         }
     );
     s.set_accept_handler(
-        [&](con_t& ep) {
-            using packet_id_t = typename std::remove_reference_t<decltype(ep)>::packet_id_t;
+        [&](con_sp_t spep) {
+            using packet_id_t = con_sp_t::element_type::packet_id_t;
             std::cout << "[server] accept" << std::endl;
-            auto sp = ep.shared_from_this();
-            ep.start_session(
-                [&, sp] // keeping ep's lifetime as sp until session finished
+            spep->start_session(
+                [&, spep] // keeping ep's lifetime as sp until session finished
                 (boost::system::error_code const& ec) {
                     std::cout << "[server] session end: " << ec.message() << std::endl;
                     s.close();
                 }
             );
 
             // set connection (lower than MQTT) level handlers
-            ep.set_close_handler(
-                [&]
+            spep->set_close_handler(
+                [&, spep]
```

Comment:
I think that kind of shared_ptr capture make a resource leak.

Reply:
```suggestion
                [&, &ep = *spep]
```

And 

```cpp
                    close_proc(connections, subs, ep.shared_from_this());
```

However, this is difficult to understand. 

I think that if there is no other easy to understand solution, revert this change is better. That means give up to pass shared_ptr to accept_handler.